### ISSUE TITLE
error in cygwin build of OSI: CoinFinite.hpp:90: error: `_isnan' undeclared

### ISSUE THREAD

**Issue by svigerske** [coin-or/Osi#16]
Issue created by migration from Trac.

Original creator: craig_schmidt

Original creation time: 2006-12-05 03:32:24

Assignee: @mjsaltzman

Version: 0.95.0

Hi There,

I am trying to build Osi stable 0.95 using cygwin on Windows XP. 
This is with gcc 3.4.4, and the updated version of make 3.81
as directed here[https://projects.coin-or.org/BuildTools/wiki/current-issues] 

I did the following steps:

1. did a new install of cygwin

2. use subversion to get OSI:

`svn co http://www.coin-or.org/svn/Osi/stable/0.95 coin-Osi`

3. did ./configure -C, which worked fine

4. did make, and got this error:

```
make[3]: Entering directory `/cygdrive/d/netflix/coin/coin-Osi/Osi/src' 
if /bin/sh ../libtool --tag=CXX --mode=compile g++ -DHAVE_CONFIG_H -I. -I`cygpath -w .` -I../inc  -I`cygpath -w /cygdrive/d/netflix/coin/coin-Osi/CoinUtils/src` -I`cygpath -w /cygdrive/d/netflix/coin/coin-Osi/Osi/../CoinUtils/inc`   -O3 -fomit-frame-pointer -pipe -DNDEBUG -pedantic-errors -Wimplicit -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion   -MT OsiAuxInfo.lo -MD -MP -MF ".deps/OsiAuxInfo.Tpo" -c -o OsiAuxInfo.lo OsiAuxInfo.cpp; \
 then mv -f ".deps/OsiAuxInfo.Tpo" ".deps/OsiAuxInfo.Plo"; else rm -f ".deps/OsiAuxInfo.Tpo"; exit 1; fi
 g++ -DHAVE_CONFIG_H -I. -I. -I../inc -Id:\\netflix\\coin\\coin-Osi\\CoinUtils\\src -Id:\\netflix\\coin\\coin-Osi\\CoinUtils\\inc -O3 -fomit-frame-pointer -pipe -DNDEBUG -pedantic-errors -Wimplicit -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -MT OsiAuxInfo.lo -MD -MP -MF .deps/OsiAuxInfo.Tpo -c OsiAuxInfo.cpp -o OsiAuxInfo.o
In file included from d:/netflix/coin/coin-Osi/CoinUtils/src/CoinMessageHandler.hpp:15,
                 from OsiSolverInterface.hpp:9,
                 from OsiAuxInfo.cpp:11:
d:/netflix/coin/coin-Osi/CoinUtils/src/CoinFinite.hpp: In function `bool CoinIsnan(double)':
d:/netflix/coin/coin-Osi/CoinUtils/src/CoinFinite.hpp:90: error: `_isnan' undeclared (first use this function)
d:/netflix/coin/coin-Osi/CoinUtils/src/CoinFinite.hpp:90: error: (Each undeclared identifier is reported only once for each function it appears in.)
make[3]: *** [OsiAuxInfo.lo] Error 1
make[3]: Leaving directory `/cygdrive/d/netflix/coin/coin-Osi/Osi/src'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/cygdrive/d/netflix/coin/coin-Osi/Osi/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/cygdrive/d/netflix/coin/coin-Osi/Osi'
make: *** [all-recursive] Error 1
```

Any suggestions?

Thanks,
Craig Schmidt

**Comment by svigerske** [comment]
Comment by Stu created at 2006-12-08 01:47:44

I've been having the same error when trying to build FlopC++ which includes Osi as an external

**Comment by svigerske** [comment]
Comment by Stu created at 2006-12-08 04:32:18

Replying to [comment:1 Stu]:
I have determined that if I change/hack the definition of MY_C_ISNAN to std::isnan i can get it to compile though that does assume I have <cmath>

Have not got the unit tests working though
Copied this ticket to https://projects.coin-or.org/CoinUtils/ticket/11

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2006-12-11 22:25:04

As of rev 877, I still see a similar problem. It looks to me like the problem is changes to BuildTools, which are incorporated in CoinUtils 1.0.4 but not in Clp 1.3.3 or DyLP 1.1.0. If I configure with --enable-doscompile, all is copacetic. Without --enable-doscompile, there are lots of interesting warnings during configuration, followed by (in my case) compilation failure in clp. The problem is that -mno-cygwin is mandatory in the Clp and DyLP releases, but the CoinUtils release incorporates the version of BuildTools (with --enable-doscompile) that makes it optional. Values stored in config.cache during CoinUtils configuration without -mno-cygwin are thus inappropriate during Clp and DyLP configuration with -mno-cygwin. The correct fix is for the Clp and DyLP project maintainers to generate updated releases.

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2006-12-11 22:25:04

Changing component from Osi Base Class to Osi Misc.

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2006-12-11 22:25:04

Changing keywords from "" to "BuildTools mismatch".

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2006-12-11 22:25:04

Version changed from 0.95.0 to 0.95

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2006-12-12 07:33:02

Use DyLP release 1.1.1, it's up-to-date with respect to BuildTools 0.5.3.

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2007-02-05 23:32:37

This really should be fixed as of BuildTools/0.5.4. I'm going to declare victory.

**Comment by svigerske** [comment]
Comment by @LouHafer created at 2007-02-05 23:32:37

Resolution: fixed
